utils/src/logger.rs

```diff
@@ -0,0 +1,58 @@
+use opentelemetry_appender_tracing::layer::OpenTelemetryTracingBridge;
+use opentelemetry_otlp::LogExporter;
+use opentelemetry_sdk::{Resource, logs::SdkLoggerProvider};
+use tracing::Level;
+use tracing_subscriber::{EnvFilter, layer::SubscriberExt, util::SubscriberInitExt};
+
+pub fn init_tracing(
+    verbosity_level: u8,
+    env_filter: Option<impl Into<EnvFilter>>,
+    otel_resource: Resource,
+    export_telemetry: bool,
+) {
+    let level = match verbosity_level {
+        1 => Level::ERROR,
+        2 => Level::WARN,
+        3 => Level::INFO,
+        4 => Level::DEBUG,
+        _ => Level::TRACE,
+    };
+    // TODO: edge-case, the u8 is 0 because it wasn't passed from cli.v
+    // if verbosity_level == 0 {
+    //     tracing::subscriber::set_global_default(tracing_subscriber::fmt().finish());
+    //     return;
+    // }
```

Comment:
Looks like we would want this, otherwise users could inadvertently end up with trace level logging.

Reply:
Yes, that should be handled, but right now this only happens when you call the function with custom arguments and not pass the cli.v - but especially since this is a crate, you're right that the case should be handled or at least throw an error